### PR TITLE
Mark queue retry processor as deprecated.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/1721

### CODE REVIEW

**Comment by pavolloffay** [comment]
Perhaps this PR should also remove https://github.com/open-telemetry/opentelemetry-collector/blob/1e65674799c8a543e1bc216b8e71de24a374991d/exporter/jaegerexporter/factory.go#L43 from the factories.

**Comment by bogdandrutu** [comment]
@pavolloffay not sure, I don't want to break current configs for the user. So just printing a log message is good I think.

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Let's start with a warning for now. We'll keep it for a while (a few months) and can then think about removing it completely.

File: processor/queuedprocessor/README.md [hunk 1]
```diff
@@ -1,47 +1,3 @@
...
-  queued_retry/example:
-    backoff_delay: 5s
-    num_workers: 2
-    queue_size: 10
-    retry_on_failure: true
-```
-
-Refer to [config.yaml](./testdata/config.yaml) for detailed
-examples on using the processor.
+QueuedRetry processor is deprecated. Use exporter queued retry config.
```
Comment (nilebox) [inline processor/queuedprocessor/README.md:3]: There should have been a link to an example of how to use it in the exporter config.

Comment (bogdandrutu) [reply processor/queuedprocessor/README.md:3]: I can review a PR :)

**Pull request merged**